test/e2e/plain_provisioner_test.go

```diff
@@ -351,116 +351,107 @@ var _ = Describe("plain provisioner bundle", func() {
 
 	When("Bundles are backed by a git repository", func() {
 		var (
-			bundles []*rukpakv1alpha1.Bundle
-			ctx     context.Context
+			ctx      context.Context
+			gitLabel = map[string]string{"git-e2e-test": "true"}
 		)
 		BeforeEach(func() {
 			ctx = context.Background()
+		})
 
-			By("creating the git based Bundles")
-			bundles = []*rukpakv1alpha1.Bundle{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						GenerateName: "combo-git-commit",
-					},
-					Spec: rukpakv1alpha1.BundleSpec{
-						ProvisionerClassName: plainProvisionerID,
-						Source: rukpakv1alpha1.BundleSource{
-							Type: rukpakv1alpha1.SourceTypeGit,
-							Git: &rukpakv1alpha1.GitSource{
-								Repository: "https://github.com/exdx/combo-bundle",
-								Ref: rukpakv1alpha1.GitRef{
-									Commit: "9e3ab7f1a36302ef512294d5c9f2e9b9566b811e",
-								},
+		AfterEach(func() {
+			list := &rukpakv1alpha1.BundleList{}
+			err := c.List(ctx, list, &client.ListOptions{LabelSelector: labels.SelectorFromSet(gitLabel)})
+			Expect(err).To(BeNil())
+
+			for _, b := range list.Items {
+				err := c.Delete(ctx, &b)
+				Expect(err).To(BeNil())
+			}
```

Comment:
Looking at this again: I think I'm torn between approving this PR so we can move onto other work, vs. second guessing this implementation. This still feels somewhat hacky, and I'd still think we'd need to move to a model that has the ability to enact a BeforeEach/AfterEach setup s.t. we don't need to perform list queries and $somewhat blindly delete a setup of Bundles that may, or may not, exist on cluster after these various tests.

Thoughts? Maybe this comes back how careful we need to be about the potential for non-deterministic tests, or maybe OLM's e2e testing suite has scarred me for life :shrug:.